src/crystal/system/unix/file_descriptor.cr

```diff
@@ -181,4 +181,14 @@ module Crystal::System::FileDescriptor
     io.sync = true
     io
   end
+
+  def self.write_at(fd, buffer, offset)
+    bytes_written = LibC.pwrite(fd, buffer, buffer.size, offset)
+
+    if bytes_written == -1
+      raise IO::Error.from_errno("Could not write to file")
+    end
+
+    bytes_written
+  end
```

Comment:
This should be an instance method and should probably be named `unbuffered_write_at`, more in line with `unbuffered_write`. Also, it should use `evented_write` to handle the fact that this write might be async.

For reference:

```cr
  private def unbuffered_write(slice : Bytes)
    evented_write(slice, "Error writing file") do |slice|
      LibC.write(fd, slice, slice.size).tap do |return_code|
        if return_code == -1 &amp;&amp; Errno.value == Errno::EBADF
          raise IO::Error.new "File not open for writing"
        end
      end
    end
  end
```